.github/workflows/_check_clang_static_analyzer.yml

```diff
@@ -29,24 +38,32 @@ jobs:
       image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
 
     steps:
-      - name: Clone
+      - name: Clone ledger-app-workflows repository
+        uses: actions/checkout@v4
+        with:
+          repository: LedgerHQ/ledger-app-workflows
+          path: ./ledger-app-workflows
+          ref: ${{ inputs.ledger-app-workflows_ref }}
+
+      - name: Clone app repository
         uses: actions/checkout@v4
         with:
+          repository: ${{ inputs.app-repository }}
+          path: app-repository
           submodules: recursive
 
       - name: Run Clippy
         if: ${{ inputs.is_rust == 'true'}}
         run: |
-          cd ${{ inputs.relative_app_directory }} && \
-          DEVICE_NAME="$(echo ${{ matrix.device }} | sed 's/nanosp/nanosplus/')" && \
-          cargo +$RUST_NIGHTLY clippy --target ${DEVICE_NAME} -- -Dwarnings
+            ./ledger-app-workflows/scripts/check_all.sh -r -c clang -t ${{ matrix.device }} \
+              -a ./app-repository -b ${{ inputs.relative_app_directory }}
```

Comment:
`clang ` arg seems not to be the best wording for rust app

Reply:
Sure. Fixed!